requirements/core.txt

```diff
@@ -1,4 +1,4 @@
 # Core dependencies.
 
 scitools-iris>=3.0.2
-eccodes-python
+eccodes>=1.6.1
```

Comment:
See #357 - I'm pulling this change into this PR even though it's already on `main`, just for consistancy.

Reply:
I don't understand the need for this.
**_Why_** is it on main? 
AFAICT, we have changed the name there, but _not_ pinned the version.

And surely, the new code will anyway still work in the "old world" ??

Reply:
Nope. The changes here don't work in the old world, hence the need for a minimum pin.

Plus, on PyPI, ECMWF decided to rename the package.

Reply:
Additionally, see this [comment](https://github.com/SciTools/iris-grib/pull/361/files#r1395419311) regarding the breaking change in the `eccodes` API 😞